Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -228,6 +229,12 @@ WarpX::AddSpaceChargeFieldLabFrame ()
     }
 #endif
     SyncRho(); // Apply filter, perform MPI exchange, interpolate across levels
+#ifndef WARPX_DIM_RZ
+    for (int lev = 0; lev <= max_level; lev++) {
+        // Reflect density over PEC boundaries, if needed.
+        ApplyRhofieldBoundary(lev, rho_fp[lev].get());
+    }
+#endif
```

Comment:
Here is another instance of the RZ and Cartesian boundary handling being treated differently - in RZ the boundary is handled before `SyncRho()` but in Cartesian it is done afterwards since filtering is done in `SyncRho()`.